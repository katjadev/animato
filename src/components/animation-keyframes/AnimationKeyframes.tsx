import { FC } from 'react'
import { AnimationGroup } from '@animato/types'
import { useEditorState } from '@animato/context/EditorContext/EditorContextProvider'
import AnimationKeyframesItem from '@animato/components/animation-keyframes-item/AnimationKeyframesItem'
import styles from './AnimationKeyframes.module.css'

interface AnimationKeyframesProps {
  element: AnimationGroup;
}

const AnimationKeyframes: FC<AnimationKeyframesProps> = ({ element }) => {
  const { state: { collapsedAnimations, selectedElementIds } } = useEditorState()
  const collapsed = collapsedAnimations.includes(element.id)
  const selected = selectedElementIds.includes(element.id)

  return (
    <>
      <div className={`${styles.keyframesEl} ${selected ? styles.selected : ''}`} />
      {!collapsed && (
        <>
          {element.animations.map((animation) => (
            <div key={animation.id} className={styles.keyframes}>
              {animation.keyframes.map((keyframe, index) => (
                <AnimationKeyframesItem 
                  key={index}
                  keyframe={keyframe}
                  prevKeyframe={animation.keyframes[index - 1]}
                />
              ))}
            </div>
          ))}
        </>
      )}
    </>
  )
}

export default AnimationKeyframes