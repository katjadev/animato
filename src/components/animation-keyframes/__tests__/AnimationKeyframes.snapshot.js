import { render } from '@testing-library/react'
import AnimationKeyframes from '../AnimationKeyframes'

const TEST_ELEMENT = {
  id: 'test-id',
  title: 'test',
  animations: [
    { 
      id: '1', 
      keyframes: [
        { time: 0 }, 
        { time: 1000 },
        { time: 2000 },
      ],
    },
  ],
}

describe('AnimationKeyframes', () => {
  it('renders correctly', () => {
    const { container } = render(<AnimationKeyframes element={TEST_ELEMENT} selected={false} />)
    
    expect(container).toMatchSnapshot()
  })

  it('enders correctly when selected', () => {
    const { container } = render(<AnimationKeyframes element={TEST_ELEMENT} selected={true} />)

    expect(container).toMatchSnapshot()
  })
})
