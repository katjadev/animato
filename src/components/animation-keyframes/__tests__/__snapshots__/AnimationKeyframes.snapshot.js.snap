// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimationKeyframes enders correctly when selected 1`] = `
<div>
  <div
    class="keyframesEl selected"
  />
  <div
    class="keyframes"
  >
    <button
      aria-label="Keyframe: 0 milliseconds"
      class="keyframe"
      style="left: 16px;"
    />
    <button
      aria-label="Keyframe: 1000 milliseconds"
      class="keyframe"
      style="left: 176px;"
    />
    <div
      class="keyframeLine"
      style="left: 16px; width: 160px;"
    />
    <button
      aria-label="Keyframe: 2000 milliseconds"
      class="keyframe"
      style="left: 336px;"
    />
    <div
      class="keyframeLine"
      style="left: 176px; width: 160px;"
    />
  </div>
</div>
`;

exports[`AnimationKeyframes renders correctly 1`] = `
<div>
  <div
    class="keyframesEl "
  />
  <div
    class="keyframes"
  >
    <button
      aria-label="Keyframe: 0 milliseconds"
      class="keyframe"
      style="left: 16px;"
    />
    <button
      aria-label="Keyframe: 1000 milliseconds"
      class="keyframe"
      style="left: 176px;"
    />
    <div
      class="keyframeLine"
      style="left: 16px; width: 160px;"
    />
    <button
      aria-label="Keyframe: 2000 milliseconds"
      class="keyframe"
      style="left: 336px;"
    />
    <div
      class="keyframeLine"
      style="left: 176px; width: 160px;"
    />
  </div>
</div>
`;
